phase3/Kconfig

```diff
@@ -40,4 +40,10 @@ config phase3.weave_net
 	  Should we deploy weave-net? You likely only want this if using kubeadm for phase2.
 endif
 
+config phase3.cni
+	string "CNI plugin to use"
+	default weave
```

Comment:
This is a little tricky. The default phase2.provider is ignition, which doesn't need a CNI provider (and installing one might be harmful?), so I think we need a consistent defaulting strategy to keep from confusing users. There are a few options I can think of here:

1. Just default to empty (or whatever the string is to disable it entirely), and mention in the help string that it's recommended to set only if phase2.provider = kubeadm.
2. Use kconfig's built-in directives to if/then/else this and have different defaults based on what phase2.provider is set to, or even omit it entirely if ignition isn't being used. For an example of this, see how [phase2.kubeadm.version](https://github.com/kubernetes/kubernetes-anywhere/blob/master/phase2/Kconfig#L28) only appears if kubeadm is already selected.

In either case, since you're adding support for flannel, I would mention in the help string what the valid options are so someone configuring a new cluster via `make menuconfig` even knows that flannel is an option.